docs/guides/server/logging.adoc

```diff
@@ -71,15 +74,90 @@ To enable log handlers, enter the following command:
 
 <@kc.start parameters="--log=\"<handler1>,<handler2>\""/>
 
-The available handlers are
-<@profile.ifCommunity>
-`console`, `file`, `syslog`.
-</@profile.ifCommunity>
-<@profile.ifProduct>
-`console`, `file` and `syslog`.
-</@profile.ifProduct>
+The available handlers are:
+
+<@opts.expectedValues option="log"/>
+
 The more specific handler configuration mentioned below will only take effect when the handler is added to this comma-separated list.
 
+== Specify log level for each handler
+
+The `log-level` property specifies the global root log level and levels for selected categories.
+However, a more fine-grained approach for log levels is necessary to comply with the modern application requirements.
+
+To set log levels for particular handlers, properties in format `log-<handler>-level` (where `<handler>` is available log handler) were introduced.
+
+It means properties for log level settings look like this:
+
+* `log-console-level` - Console log handler
+* `log-file-level` - File log handler
+* `log-syslog-level` - Syslog log handler
+
+NOTE: The `log-<handler>-level` properties are available only when the particular log handlers are enabled.
+More information in log handlers settings below.
+
+Accepted log levels are specified in <<Log levels>> section, and *must be in lowercase*.
+
+=== General principle
+
+It is necessary to understand that setting the log levels for each particular handler *does not override the root level* specified in the `log-level` property.
+It specifies the minimal allowed level for logs shown in the output.
+It respects levels specified in the `log-level` property, which creates a minimal log level bar for log records.
+
+Specifically, when an arbitrary log level is specified for the handler, it does not mean the log records with the log level will be present in the output.
+The root log level needs to be assessed as well, as it determines the minimal level for exposed log records.
```

Comment:
Even though I accepted your suggestion, I thought about it once again, and I think the first idea was better. I understand it might be confusing a bit, but it depends on the PoV - verbosity x severity levels. 

For multiple(all I know) log levels representations, the levels are sorted descending based on the severity. All docs show the table of log levels also based on the severity. So, IMHO, it's more intuitive to say the `trace` is minimal instead of `fatal` is minimal - so, **severity**, no verbosity. 

java.util.logging.Level:
```
SEVERE = 1000
WARNING = 900
....
FINEST = 300
``` 

JBoss LogManager:
```
FATAL = 1100
ERROR = 1000
TRACE = 400
```

Also the Quarkus sets the minimal level (`quarkus.log.min-level`) to `debug` (based on severity).

So I'd propose using the 'minimal' allowed level as it's more natural for users imagine the log level table, where the minimal item is `TRACE`/`FINEST`. 

@vmuzikar Is it ok? Or should I make it more clear - explicitly mention the terminology?

Reply:
That's a good point we're basically aligning with Quarkus on the wording but IMHO it does not necessarily mean it's correct. :) What I find confusing is that AFAIK there's an established term "increase log level" but at the same time we're saying e.g. `debug` is less than `error`.

But maybe we could find some middle ground here to make it clear. Instead of minimal/maximal we'd either say most/least verbose, or most/least severe. WDYT?

Reply:
Fair enough; I'll do. Thanks!

Reply:
@vmuzikar In the end, I used some kind of hybrid approach to properly reference the log-level bars. I hope it's okay. Please review.